File: tests/vb/vb_linq_distinct/linq_distinct.vb
' vybe-test: vb/vb_linq_distinct/linq_distinct
' origin: languages/vb/tests/vb/test_vb_linq_distinct.rs

Imports System.Linq

Module M
    Sub Main()
        Dim numbers = {1, 2, 2, 3, 3, 3}
        
        Dim query = (From n In numbers Select n).Distinct()
        
        For Each n In query
            Console.WriteLine(n)
        Next
    End Sub
End Module

File: tests/csharp/linq_lambdas/linq_distinct.cs
// vybe-test: csharp/linq_lambdas/linq_distinct
// origin: languages/csharp/tests/csharp/test_linq_lambdas.rs

using static __Harness;

var nums = new List<int> { 1, 2, 2, 3, 3, 3, 4 }
;
var distinct = nums.Distinct().ToList();
__P((distinct.Count).ToString());
__Check("4");

public static class __Harness {
    public static string __buf = "";
    public static void __P(string s) { __buf = __buf + s + "\n"; }
    public static void __Pr(string s) { __buf = __buf + s; }
    public static void __Check(string want) {
        if (__buf != want && __buf != want + "\n") {
            Console.WriteLine("FAIL: want [" + want + "] got [" + __buf + "]");
            throw new Exception("assertion failed");
        }
    }
}
